scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+
+    private String sftpUrlKey;
+
+    private String sftpPassword;
+
+    private static final String SFTP_LOCAL_RELATIVE_PATH_ARG = "sftpLocalRelativePath";
+
+    @Override
+    public void init(Map<String, Serializable> args) throws Exception {
+        if (args.containsKey("sftpHostname")) {
+            sftpHostname = args.get("sftpHostname").toString();
+        }
+        if (args.containsKey("sftpPort")) {
+            sftpPort = Integer.parseInt(args.get("sftpPort").toString());
+        }
+        if (args.containsKey("sftpMode")) {
+            sftpMode = args.get("sftpMode").toString();
+        }
+        if (args.containsKey("sftpExtractArchive")) {
+            sftpExtractArchive = Boolean.parseBoolean(args.get("sftpExtractArchive").toString());
+        }
+        if (args.containsKey(SFTP_LOCAL_RELATIVE_PATH_ARG) &&
+            !args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString().isEmpty()) {
+            sftpLocalRelativePath = args.get(SFTP_LOCAL_RELATIVE_PATH_ARG).toString();
+        } else {
+            //we throw an exception tp prevent transferring all the contents of the global space.
+            if (sftpMode.equals("PUT")) {
+                throw new IllegalArgumentException("Please specify a local relative path. Empty value is not allowed.");
+            }
+            //default value is getlocalspace() because it will always be writable and moreover can be used to transfer files to another data space(global, user)
+            sftpLocalRelativePath = getLocalSpace();
+        }
+        if (args.containsKey("sftpRemoteRelativePath")) {
+            sftpRemoteRelativePath = args.get("sftpRemoteRelativePath").toString();
+        }
+
+        if (args.containsKey("sftpUsername")) {
+            sftpUsername = args.get("sftpUsername").toString();
+        }
+
+        if (sftpUsername == null) {
+            throw new IllegalArgumentException("Username is required to access an SFTP server");
+        }
+
+        // This key is used for logs and for getting the password from 3rd party credentials.
+        sftpUrlKey = "sftp://" + sftpUsername + "@" + sftpHostname + ":" + sftpPort;
+
+        sftpPassword = getThirdPartyCredential(sftpUrlKey);
+
+        if (sftpPassword == null) {
+            throw new IllegalArgumentException("Please add your sftp password to 3rd-party credentials under the key :\"" +
+                                               sftpUrlKey + "\"");
+        }
+    }
+
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, JSchException, SftpException {
+
+        List<String> filesRelativePathName = null;
+
+        ChannelSftp channelsftp = null;
+        Session session = null;
+        Channel channel = null;
+
+        try {
+            JSch jsch = new JSch();
+            session = jsch.getSession(sftpUsername, sftpHostname, sftpPort);
+            session.setPassword(sftpPassword);
+            Properties config = new Properties();
```

Comment:
You could extract the three lines of SSH configuration in a dedicated private method (in case we need to add more config for exmaple).